test/unit/node/settings.test.ts

```diff
@@ -0,0 +1,65 @@
+import { logger } from "@coder/logger"
+import { promises as fs } from "fs"
+import { SettingsProvider, CoderSettings } from "../../../src/node/settings"
+import { clean, mockLogger, tmpdir } from "../../utils/helpers"
+
+describe("settings", () => {
+  describe("with invalid JSON in settings file", () => {
+    const testName = "settingsInvalidJSON"
+    let testDir = ""
+    let mockSettingsFile = "coder.json"
+    let pathToMockSettingsFile = ""
+
+    beforeEach(async () => {
+      mockLogger()
+      await clean(testName)
```

Comment:
This is completely up to personal preference but using a single temporary directory per suite is a little easier since you would not need to duplicate this code for each test (this is how the other tests are doing it I believe).

If we want to do it this way we could move it to a `beforeEach` in the top-level describe (can also combine the logger mock if desired).

Reply:
I'm all for less code so I'll do this. Thanks for the recommendation! I was worried expecting mock calls in each might cause conflicts, but I think as long as I clear the mocks `afterEach` it should be fine.